extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts

```diff
@@ -107,16 +108,31 @@ export class MssqlObjectExplorerNodeProvider extends ProviderBase implements azd
 			if (node) {
 				expandResult.errorMessage = node.getNodeInfo().errorMessage;
 				let children = await node.getChildren(true);
-				if (children) {
-					expandResult.nodes = children.map(c => c.getNodeInfo());
-					// There is only child returned when failure happens
-					if (children.length === 1) {
-						let child = children[0].getNodeInfo();
-						if (child && child.nodeType === constants.MssqlClusterItems.Error) {
-							expandResult.errorMessage = child.label;
-							expandResult.nodes = [];
+				if(children && children.length > 0)
+				{
+					let child = children[0].getNodeInfo();
+					// Only child returned when failure happens : When failed with 'Unauthorized' error, prompt for password.
+					if(children.length === 1 && child && child.nodeType === constants.MssqlClusterItems.Error)
+					{
+						if (children[0].errorStatusCode === 401)
+						{
+							//Prompt for password
+							let password: string = await this.promptPassword(localize('prmptPwd', 'Please provide the password to connect to HDFS:'));
+							if(password && password.length > 0){
+								session.sqlClusterConnection.updatePassword(password);
+								node.updateFileSource(session.sqlClusterConnection);
+								children = await node.getChildren(true);
+								child = children[0].getNodeInfo();
```

Comment:
You don't know if children will have 1 or more nodes here - please check it's non-null and has length > 1